### PR TITLE
Add Sorted method

### PR DESCRIPTION
**Pull request by Frassle**
This is a set variant on `slices.Sort` that makes it easy to get a sorted slice. This makes it easy to get the elements of the set into a sorted slice. Quite handy when using sets to build up unique elements but then needing a well defined iteration order.

### CODE REVIEW

**Comment by Frassle** [comment]
A helpful wrapper around https://pkg.go.dev/slices@go1.21.1#Sort. I could add SortedFunc as well if wanted to wrap https://pkg.go.dev/slices@go1.21.1#SortFunc.

**Comment by deckarep** [comment]
@Frassle - thanks for your contribution. This seems like a nice and convenient helper func. I’ll look it over in detail along with your other posted issue likely tomorrow.

Thanks for your interest and contribution!

**Comment by Frassle** [comment]
Noticed from your CI run that "slices" and "cmp" are only in go 1.21 so I've moved this to a new file that can be conditional by version. Won't break existing users, but anyone wanting to use sorted will need to be on 1.21.

**Comment by Frassle** [comment]
Hey @deckarep just to check in if you've had a chance to consider this yet.

File: sorted_test.go [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+		if len(sorted) != set.Cardinality() {
+			t.Errorf("Length of slice is not the same as the set. Expected: %d. Actual: %d", set.Cardinality(), len(sorted))
+		}
+
+		if sorted[0] != "apple" {
+			t.Errorf("Element 0 was not equal to apple: %s", sorted[0])
+		}
+		if sorted[1] != "banana" {
+			t.Errorf("Element 0 was not equal to banana: %s", sorted[1])
+		}
```
Comment (funwithbots) [inline sorted_test.go:49]: nit. Error should refer to `Element 1...` same issue for next `Element 2...`

**Comment by Frassle** [comment]
Thanks! And no worries about the time, I know how easy it is to lose track of things like this.

**Pull request merged**